event/parsing/provide.go

```diff
@@ -0,0 +1,87 @@
+package parsing
+
+import (
+	"github.com/go-kit/kit/log"
+	"github.com/xmidt-org/arrange"
+	"github.com/xmidt-org/bascule/acquire"
+	"github.com/xmidt-org/webpa-common/logging"
+	"go.uber.org/fx"
+)
+
+// CodexConfig determines the auth and address for connecting to the codex cluster
+type CodexConfig struct {
+	Address string
+	Auth    AuthAcquirerConfig
+}
+
+type AuthAcquirerConfig struct {
+	JWT   acquire.RemoteBearerTokenAcquirerOptions
+	Basic string
+}
+
+// ParsersIn brings together all of the different types of parsers that glaukos uses
+type ParsersIn struct {
+	fx.In
+	BootTimeParser Parser `name:"bootTimeParser"`
+	MetadataParser Parser `name:"metadataParser"`
+}
+
+// ParsersOut provides the parsers for other uber fx components to use
+type ParsersOut struct {
+	fx.Out
+	BootTimeParser Parser `name:"bootTimeParser"`
+	MetadataParser Parser `name:"metadataParser"`
+}
+
+// Provide bundles everything needed for setting up all of the event objects
+// for easier wiring into an uber fx application.
+func Provide() fx.Option {
+	return fx.Options(
+		ProvideEventMetrics(),
+		fx.Provide(
+			arrange.UnmarshalKey("codex", CodexConfig{}),
+			func(in MetricsIn) MetadataParser {
+				return MetadataParser{
+					MetadataFields: in.MetadataFields,
+				}
+			},
+			func(logger log.Logger, metricsIn MetricsIn, codexConfig CodexConfig) BootTimeCalc {
+				codexAuth, err := determineCodexTokenAcquirer(logger, codexConfig)
+				if err != nil {
+					logging.Error(logger).Log(logging.MessageKey(), "failed to create acquirer", "error", err)
+				}
+
+				return BootTimeCalc{
+					BootTimeHistogram: metricsIn.BootTimeHistogram,
+					Logger:            logger,
+					Address:           codexConfig.Address,
+					Auth:              codexAuth,
+				}
+			},
+			func(calc BootTimeCalc, mp MetadataParser) ParsersOut {
+				return ParsersOut{
+					BootTimeParser: calc,
+					MetadataParser: mp,
+				}
+			},
```

Comment:
With my suggested change above, this code isn't necessary.  Each function that is creating the Parsers can be `Annotated` to the same group name.
```suggestion
```